code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -121,7 +122,13 @@
 
 /mob/living/silicon/robot/proc/is_component_functioning(module_name)
 	var/datum/robot_component/C = components[module_name]
-	return C && C.installed == 1 && C.toggled && C.is_powered()
+	return C && C.installed == 1 && C.toggled && C.is_powered() && !C.component_disabled
+
+/mob/living/silicon/robot/proc/disable_component(module_name, duration)
+	 var/datum/robot_component/D = get_component(module_name)
+	 D.component_disabled = 1
+	 spawn(duration)
+	 		D.component_disabled = 0
```

Comment:
If someone could please explain to me, does this actually have an extra indent? I swear there is no extra indent in my actual repo.
If you need proof: https://gyazo.com/0190f71532e628493833b1f4e06faa3d